src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -5287,12 +5287,16 @@ protected synchronized void drain(boolean isFinalShutdown) throws IOException, I
             CommitLog.instance.shutdownBlocking();
 
             // wait for miscellaneous tasks like sstable and commitlog segment deletion
-            ScheduledExecutors.nonPeriodicTasks.shutdown();
-            if (!ScheduledExecutors.nonPeriodicTasks.awaitTermination(1, MINUTES))
-                logger.warn("Unable to terminate non-periodic tasks within 1 minute.");
-
             ColumnFamilyStore.shutdownPostFlushExecutor();
```

Comment:
Why did you move the shutdownPostFlushExecutor ahead of the refactored non-periodic executor termination?

I couldn't find any tasks that obviously scheduled on the post flush executor that should still be running - it looks mostly related to the commit log and column family truncation which should not be possible on the node at this point in the drain.

So I think this is safe, but was wondering if there was a reason for it or it was an incidental change.

Reply:
Well, the movement was done purely by looking at it - I similarly think nothing is scheduling on that executor. There is stuff related to flushing commit logs etc. There seems to be nothing which would schedule stuff for ColumnFamilyStore to these executors.

This is again something I consider to be the flaw of the previous solution. I think that all logic should be "stopped" first before shutting down all executors. I would say that the subsystem should clean up after itself first and then we are safer to shut down everything else rather than thinking that we submitted a task to these executors on which ColumnFamilyStore depends. The tests / pipeline proves me right that, I do not see any regression.